### DISCUSSION TITLE
How can I use NuGet package Microsoft.WindowsPackageManager.Utils in C#

### DISCUSSION THREAD

**Discussion by SlimRG** [microsoft/winget-cli#3343] [General] [answered]
I can't find any info about how can I use winget nuget package inside my C# project.
What functions or class it has?
Can you give some more information for developers?
Thanks!

**Comment by SlimRG** [comment]
And what?
No changes after year....

**Comment by JohnMcPMS** [comment]
The relevant C header is here: [WinGetUtil.h](https://github.com/microsoft/winget-cli/blob/master/src/WinGetUtil/WinGetUtil.h)
 
The utils nuget package is not intended for public use, so there is not documentation beyond the code.

**Comment by SlimRG** [comment]
> The relevant C header is here: [WinGetUtil.h](https://github.com/microsoft/winget-cli/blob/master/src/WinGetUtil/WinGetUtil.h)
> 
> The utils nuget package is not intended for public use, so there is not documentation beyond the code.

I asked about C# nuget package

**Comment by JohnMcPMS** [comment, marked as answer]
It is a native nuget package, so you would have to PInvoke these methods.
